components/director/internal/domain/labeldef/repository.go

```diff
@@ -0,0 +1,91 @@
+package labeldef
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/internal/persistence"
+	"github.com/pkg/errors"
+)
+
+const (
+	tableName = "public.label_definitions"
+)
+
+type repo struct {
+	conv Converter
+}
+
+func NewRepository(conv Converter) *repo {
+	return &repo{conv: conv}
+}
+
+func (r *repo) Create(ctx context.Context, def model.LabelDefinition) error {
+	db, err := persistence.FromCtx(ctx)
+	if err != nil {
+		return err
+	}
+
+	entity, err := r.conv.ToEntity(def)
+	if err != nil {
+		return errors.Wrap(err, "while converting Label Definition to insert")
+	}
+
+	columns := []string{"id", "tenant_id", "key"}
+	if entity.SchemaJSON.Valid {
+		columns = append(columns, "schema")
+	}
+	values := r.prefixEveryWithColon(columns)
+
+	_, err = db.NamedExec(fmt.Sprintf("insert into %s (%s) values(%s)", tableName, strings.Join(columns, ","), strings.Join(values, ",")), entity)
+	if err != nil {
+		return errors.Wrap(err, "while inserting Label Definition")
+	}
+	return nil
+}
+
+func (r *repo) prefixEveryWithColon(in []string) []string {
+	out := make([]string, 0)
+	for _, elem := range in {
+		out = append(out, ":"+elem)
+	}
+	return out
+}
+
+func (r *repo) GetByKey(ctx context.Context, tenant string, key string) (*model.LabelDefinition, error) {
+	db, err := persistence.FromCtx(ctx)
+	if err != nil {
+		return nil, err
+	}
+	dest := Entity{}
+
+	q := fmt.Sprintf("select * from %s where tenant_id=$1 and key=$2 ", tableName)
+
+	err = db.Get(&dest, q, tenant, key)
+	switch {
+	case err == sql.ErrNoRows:
+		return nil, nil
```

Comment:
For later: We need to discuss common approach for the "Not found" situation. I think the repository should return an error, but the one of custom type (our own "Entity not found" error).